Automattic/wp-e2e-tests
# Remove this.bailSuite

This is a huge PR 

Removes all the references to `test.it` and replaces these with `step` based upon using mocha-steps 👍 

Removes all references to anything `test.` and uses standard mocha blocks now as we don't use the promise manager

Removes our dependency on a custom version of mocha we host at https://github.com/Automattic/mocha

**Testing/Review Instructions**

1. Make sure everything looks ok 
2. Make sure no references to test blocks
3. Make sure package looks correct
